keps/sig-multicluster/1645-multi-cluster-services-api/specification.md

```diff
@@ -0,0 +1,263 @@
+
+Cluster ID / `<clusterid>` = the cluster id stored in the `id.k8s.io ClusterProperty` as described in [KEP-2149: ClusterId for ClusterSet identification](../2149-clusterid/README.md). Though this can be any valid DNS label, the recommended value is  a kube-system namespace uid ( such as `721ab723-13bc-11e5-aec2-42010af0021e`). For ease of KEP readability, this document uses human readable names `clusterA` and `clusterB` to represent the cluster IDs of two clusters in a ClusterSet.
+
+
+### 2.2 - Record for Schema Version
+
+Following the existing specification, clusters implementing multicluster DNS will contain an additional `TXT` record responding with the semantic version of the DNS schema used for the multi cluster DNS `<zone>`, also known in this specification as `<clustersetzone>`.
+
+- Question Example:
+  - `dns-version.clusterset.local. IN TXT`
+- Answer Example:
+  - `dns-version.clusterset.local. 28800 IN TXT “1.0.0”`
+
+### 2.3 - Records for a Service with ClusterSetIP
+
+#### 2.3.1 - `A`/`AAAA` Record
+
+_Note: This section refers to `A` and `AAAA` record requirements. For Service objects that have dual-stack networking enabled, both `A` and `AAAA` records must be created to cover both IPv4 and IPv6 assigned ClusterSetIPs._
+
+Given a ClusterIP type Service named `<service>` in Namespace `<ns>` that has been exported via a name-mapped ServiceExport with name `<service>`, given it is accessible across the clusterset by the IP address `<clusterset-ip>`, the following records must exist.
+
+If the `<clusterset-ip>` is an IPv4 address, an `A` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `<service>.<ns>.svc.<clustersetzone>. <ttl> IN A <clusterset-ip>`
+*   Question Example
+    *   `myservice.test.svc.clusterset.local. IN A`
+*   Answer Example:
+    *   `myservice.test.svc.clusterset.local. 4 IN A 10.42.42.42`
+
+If the `<clusterset-ip>` is an IPv6 address, an `AAAA` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `<service>.<ns>.svc.<clustersetzone>. <ttl> IN AAAA <clusterset-ip>`
+*   Question Example:
+    *   `myservice.test.svc.clusterset.local. IN AAAA`
+*   Answer Example:
+    *   `myservice.test.svc.clusterset.local. 4 IN AAAA 2001:db8::1`
+
+
+#### 2.3.2 - `SRV` Records
+
+For each port in an exported Service with name `<port>` and number `<port-number>` using protocol `<proto>`, an `SRV` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `_<port>._<proto>.<service>.<ns>.svc.<clustersetzone>. <ttl> IN SRV <weight> <priority> <port-number> <service>.<ns>.svc.<zone>.`
```

Comment:
s/zone/clusterset-zone/ ?  (look at the end of line)

Reply:
Fixed in 42a1f79